### DISCUSSION TITLE
exportFile charset

### DISCUSSION THREAD

**Discussion by ziltonvasconcelos** [quasarframework/quasar#8846] [General - Components / Directives / etc]
I am using exportFile Utils from quasar

https://quasar.dev/quasar-utils/other-utils

but I cannot set charset for the csv file...any idea how?

thanks,
Zilton.

**Comment by rstoenescu** [comment]
Third parameter of exportFile(). Set it to something like: `text/plain;charset=utf-8`.

**Reply by ziltonvasconcelos** [reply]
Thanks! But whenever I open with excell did not work! Is there any thing else to do it in this case?

**Reply by jbrutcho** [reply]
I'm dealing with the same problem as you. Did you find the solution?
Thanks

**Reply by dorobon** [reply]
same here

**Comment by ziltonvasconcelos** [comment]
No solution until now!
_____

Zilton Vasconcelos

Pesquisador em Saúde Pública
Fundação Oswaldo Cruz - FIOCRUZ
Instituto Fernandes Figueira - IFF
Laboratório de Alta Complexidade - LACIFF
Departamento de Pesquisa Clínica
***@***.*** ***@***.***>
Av. Rui Barbosa, 716 - Flamengo - Cep: 22250-020 - Rio de Janeiro/RJ -
Brasil
Office: (21)2554-1919
Lab: (21)2554-1967/1968
Celular: (21)99812-7196


Em qui., 13 de mai. de 2021 às 15:17, jbrutcho ***@***.***>
escreveu:

**Comment by willagner** [comment]
I have been tried many solutions to solve this problem too, but i didn't can.

**Comment by willagner** [comment]
I tried with this 2 examples below.

This example i taked in the documentation of the Quasar.

    exportTable: function() {

      let csvContent = "data:text/csv;charset=utf-8,";
      csvContent += [
        Object.keys(this.data[0]).join(";"),
        ...this.data.map(item => Object.values(item).join(";"))
      ]
        .join("\n")
        .replace(/(^\[)|(\]$)/gm, "");
      const data = encodeURI(csvContent);
      const link = document.createElement("a");
      link.setAttribute("href", data);
      link.setAttribute("download", "teste.csv");
      link.click();
    },


This example below i found with vue export json to csv.

    export () {
      // naive encoding to csv format
      const content = [
        this.columns.map((col) => wrapCsvValue(col.label)).join(";"),
      ]
        .concat(
          this.data.map((row) =>
            this.columns
              .map((col) =>
                wrapCsvValue(
                  typeof col.field === "function"
                    ? col.field(row)
                    : row[col.field === void 0 ? col.name : col.field],
                  col.format
                )
              )
              .join(";")
          )
        )
        .join("\r\n");

      const status = exportFile("table-export.csv", content, "text/csv");

      if (status !== true) {
        this.$q.notify({
          message: "Browser denied file download...",
          color: "negative",
          icon: "warning",
        });
      }
    },


But didn't work.

**Reply by hawkeye64** [reply]
You're joining by semi-colon (;) when it should be comma (,)
CSV = Comma-Separated Values
Here is a list of common mime-types: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

**Reply by willagner** [reply]
My problem is with special characters, when i downloaded the json to csv the data is showed like this 'NÃ£o' when the correctly is Não.

![erro csv](https://user-images.githubusercontent.com/26633163/118363344-42fd7600-b56a-11eb-857e-1e0abc215257.PNG)


I tried put utf-8 to solve this problem but didn't work.
let csvContent = "data:text/csv;charset=utf-8,";

**Reply by hawkeye64** [reply]
Instead of `utf-8` you can try `utf-16` or `ISO-8859-1`...

**Reply by willagner** [reply]
I tried:

 set language pt-BR in quasar.conf.js

set utf-8 and utf-16 in index.template.html

And test the three options to encode utf 8, 16 and ISO

    export () {
          // naive encoding to csv format
          const content = [
            this.columns.map((col) => wrapCsvValue(col.label)).join(";"),
          ]
            .concat(
              this.data.map((row) =>
                this.columns
                  .map((col) =>
                    wrapCsvValue(
                      typeof col.field === "function"
                        ? col.field(row)
                        : row[col.field === void 0 ? col.name : col.field],
                      col.format
                    )
                  )
                  .join(";")
              )
            )
            .join("\r\n");
    
        const status = exportFile("table-export.csv", content, **"data:text/csv;charset=utf-16,"**);
    
        if (status !== true) {
          this.$q.notify({
            message: "Browser denied file download...",
            color: "negative",
            icon: "warning",
          });
        }
      },

**Reply by PlumaCymbidium** [reply]
`const status = exportFile("table-export.csv", "\ufeff"+content, "text/csv"); ` works for me.
